docs/how-tos/upload-data.md

```diff
@@ -0,0 +1,238 @@
+Click **Save** in the top right corner of the page.
+
+{{% /tablestep %}}
+{{% tablestep %}}
+**3. Confirm that your data uploaded**
+
+Navigate to your [**DATA** page in the Viam app](https://app.viam.com/data/view) and confirm that your data appears there.
+If you don't see your files yet, wait a few moments and refresh the page.
+
+{{% /tablestep %}}
+{{% tablestep %}}
+**4. Remove the folder path**
+
+Once the data has uploaded, navigate back to your data service config.
+You can now delete the additional path you added.
+You can also turn off **Syncing** unless you have other directories you'd like to continue to sync from.
+{{% /tablestep %}}
+{{< /table >}}
+
+## Upload data with Python
+
+You can use the Viam Python SDK's data client API [`file_upload_from_path`](/appendix/apis/data-client/#fileuploadfrompath) method to upload one or more files from your computer to the Viam cloud.
+
+{{% alert title="Note" color="note" %}}
+
+Unlike data sync, using the `file_upload_from_path` API method uploads all the data even if that data already exists in the cloud.
+In other words, it duplicates data if you run it multiple times.
+
+Also unlike data sync, this method _does not_ delete data from your device.
+
+{{% /alert %}}
+
+### Prerequisites
+
+{{< expand "Install the Viam Python SDK" >}}
+
+Install the [Viam Python SDK](https://python.viam.dev/) by running the following command on the computer from which you want to upload data:
+
+```sh {class="command-line" data-prompt="$"}
+pip install viam-sdk
+```
+
+{{< /expand >}}
+
+### Instructions
+
+{{< table >}}
+{{% tablestep link="/appendix/apis/data-client/" %}}
+**1. Write connection code**
+
+Create a Python script file in a directory of your choice and [add code to establish a connection](/appendix/apis/data-client/#establish-a-connection) from your computer to your [Viam app](https://app.viam.com) {{< glossary_tooltip term_id="location" text="location" >}} or individual {{< glossary_tooltip term_id="part" text="machine part" >}}.
+
+{{% /tablestep %}}
+{{% tablestep link="/appendix/apis/data-client/" %}}
+**2. Add a `file_upload_from_path` API call**
+
+Use the `file_upload_from_path` method to upload your data, depending on whether you are uploading one or multiple files:
+
+{{< tabs >}}
+{{% tab name="Upload a single file" %}}
+
+To upload just one file, make a call to `file_upload_from_path` according to [the data client API documentation](/appendix/apis/data-client/#fileuploadfrompath).
+The following example code could be placed inside the `main()` function, or inside a function called from `main()`:
+
+```python {class="line-numbers linkable-line-numbers"}
+await data_client.file_upload_from_path(
+  # The ID of the machine part the file should be associated with
+  part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+  # Any tags you want to apply to this file
+  tags=["cat", "animals", "brown"],
+  # Path to the file
+  filepath="/Users/Artoo/my_cat_photos/brown-cat-on-a-couch.png"
+)
```

Comment:
```suggestion
import asyncio

from viam.rpc.dial import DialOptions, Credentials
from viam.app.viam_client import ViamClient


async def connect() -> ViamClient:
    dial_options = DialOptions(
      credentials=Credentials(
        type="api-key",
        # Replace "<API-KEY>" (including brackets) with your machine's API key
        payload='<API-KEY>',
      ),
      # Replace "<API-KEY-ID>" (including brackets) with your machine's
      # API key ID
      auth_entity='<API-KEY-ID>'
    )
    return await ViamClient.create_from_dial_options(dial_options)


async def main():
    # Make a ViamClient
    viam_client = await connect()
    # Instantiate a DataClient to run data client API methods on
    data_client = viam_client.data_client
    await data_client.file_upload_from_path(
      # The ID of the machine part the file should be associated with
      part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
      # Any tags you want to apply to this file
      tags=["cat", "animals", "brown"],
      # Path to the file
      filepath="/Users/Artoo/my_cat_photos/brown-cat-on-a-couch.png"
    )

    viam_client.close()

if __name__ == '__main__':
    asyncio.run(main())
```

Reply:
hmm so steps 1 and 2 should be combined? In which case tablestep formatting probably doesn't make sense for this whole section anymore eh?

Reply:
This feels a bit unnecessarily hand-holdy to me since they need to customize the script for their use case anyway, with different tags, part ID, filepath, and API keys

Reply:
It means fewer pages they need to open. It is much nicer to copy an entire script and swap out a few things than to have to piece things together from different pages. You can have copying it be the first step and  finding the ids the second

Reply:
Okay I'm going to leave the order as-is and add this code as expanders. I think it still makes sense that way. But, I'm having a code formatting issue when I put them in expanders https://viaminc.slack.com/archives/C055G9S1W3Y/p1725493005478279